recipes/circularbuffer/all/test_package/conanfile.py

```diff
@@ -1,16 +1,27 @@
+from conan import ConanFile
+from conan.tools.build import  can_run
+from conan.tools.cmake import CMake, cmake_layout
+
 import os
-from conans import ConanFile, CMake, tools
 
-class TestConan(ConanFile):
+
+class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            bin_path = os.path.join("bin", "test_package")
+        if can_run(self):
+            bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             self.run(bin_path, run_environment=True)
```

Comment:
```suggestion
            self.run(bin_path, env="conanrun")
```
This is missing to apply VirtualRunEnv. I'm not sure it matters on this recipe but better to nail it in case someone uses this recipe as reference.

Reply:
Thank you! I have seen that in other recipes also, but was not sure if it's the same but just different, or something new required

Reply:
Run environment is removed in 2.0 and does nothing so this is required for backwards 1.x with the new generators being off by default :)